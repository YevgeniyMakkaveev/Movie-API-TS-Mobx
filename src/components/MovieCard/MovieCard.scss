@import '../../style/color.scss';

.wrapper {
 width: 100%;
 padding-top: 2em;

 .preview {
  width: 1200px;
  margin: 4em auto;
  background-color: $primary-color;
  border-radius: 4px;

  .head {
   display: flex;
   flex-direction: row;
   margin: 0 5em;
   justify-content: space-between;

   img {
    margin-top: 1em;
    height: 700px;
    width: auto;
   }

   .movie-title{
    font-size: 3em;
    color: $secondary-color;
    font-weight: 700;
   }
   .tagline{
    font-size: 2em;
    color: $secondary-color;
    font-weight: 400;
    margin-bottom: 1em;
   }
   .main-info {
    display: flex;
    width: 100%;
    flex-direction: column;
    margin-top: 2em;
    align-items: center;
   }
   .date,
   .genres {
    font-size: 1.2em;
    color: $white-text;
    margin-bottom: 1em;
   }

  }

  .body {
   color: #F4F5FF;
   padding: 2rem;
   display: flex;
   justify-content: center;
   align-items: center;
  }
 }
}

.green, .orange, .red{
 font-size: 2em;
}


.green {
 color: lightgreen;
}

.orange {
 color: darkgoldenrod;
}

.red {
 color: darkred;
}