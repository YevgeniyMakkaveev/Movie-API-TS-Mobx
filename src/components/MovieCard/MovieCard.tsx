import React from "react";

import IMovieCard from "../../types/movieCard";
import getColor from "../../unit/getColor";
import {_imgPatch} from '../../_const';
import './MovieCard.scss'

const MovieCard:React.FC<IMovieCard> =({overview,poster_path,release_date,title,vote_average,tagline,genres})=>{
console.log('загрузка')

 return <div className='wrapper' >
  <div className='preview'>
<div className='head'>
 <img src={`${_imgPatch}${poster_path}`} alt={title} />
 <div className='main-info'>
  <div><h2 className='movie-title'>{title}</h2></div>
  <div className='tagline'>{tagline} </div>
  <div className='date'>Release date: {release_date}</div>
  <div className='genres'>
  Genres:  {genres&&genres.map(genre=><span>{genre.name} </span>)}
    </div>
  <div className={getColor(+vote_average)} > Score {vote_average}</div>
 </div>
</div>
<div className='body'>
 {overview}
</div>
</div>
 </div>
}
export default MovieCard