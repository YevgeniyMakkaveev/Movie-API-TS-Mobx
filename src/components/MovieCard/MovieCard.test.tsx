import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import MovieCard from "./MovieCard";

const app = (
  <MovieCard
    id={1}
    title="Name"
    poster_path={"/none"}
    overview={"I did it for test"}
    vote_average={"8"}
    release_date="1984"
    tagline="testing"
    toggleFavorite={() => {}}
    isFavorite={true}
    genres={[
      { id: 1, name: "1" },
      { id: 2, name: "2" },
    ]}
  />
);

describe("Movie card component should work", () => {
  it("should render correctly", () => {
    render(
      <MovieCard
        id={1}
        title="Name"
        poster_path={"/none"}
        overview={"I did it for test"}
        vote_average={"8"}
        release_date="1984"
        tagline="testing"
        toggleFavorite={() => {}}
        isFavorite={true}
        genres={[
          { id: 3, name: "1" },
          { id: 4, name: "2" },
        ]}
      />
    );
    expect(screen.getByText("Name")).toBeInTheDocument();
    expect(screen.getByText("I did it for test")).toBeInTheDocument();
    expect(screen.getByText("Release date: 1984")).toBeInTheDocument();
    expect(screen.getByText("Score 8")).toBeInTheDocument();
    expect(screen.getByText("1")).toBeInTheDocument();
  });
  describe("it should be used to test snapshots", () => {
    it("should mach the snapshot", () => {
      const snap = app;
      expect(snap).toMatchSnapshot();
    });
  });
});
