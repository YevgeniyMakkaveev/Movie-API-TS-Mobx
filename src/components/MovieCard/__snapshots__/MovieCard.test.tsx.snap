// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card component should work it should be used to test snapshots should mach the snapshot 1`] = `
<MovieCard
  genres={
    Array [
      Object {
        "id": 1,
        "name": "1",
      },
      Object {
        "id": 2,
        "name": "2",
      },
    ]
  }
  id={1}
  isFavorite={true}
  overview="I did it for test"
  poster_path="/none"
  release_date="1984"
  tagline="testing"
  title="Name"
  toggleFavorite={[Function]}
  vote_average="8"
/>
`;
